### PR TITLE
Removed raising of AssertionError from 'io.fits.fitsrec.__del__' (rebased version of #4694)

### LINKED ISSUE #4690: Rice compression emits spurious text when closing
Starting with astropy version 1.1.2, closing a rice-compressed file emits a bunch of spurious text along the lines of: `Exception AttributeError: AttributeError('_coldefs',) in <bound method FITS_rec.__del__ of FITS_rec .... ignored`.

From what I understand, this means there is an exception that is trying to be raised in the `__del__` method, but you're not allowed to do that, so it emits text rather than raising the exception.

Here is a minimal example to reproduce the problem:
```
import astropy.io.fits as pyfits

# from http://fits.gsfc.nasa.gov/registry/tilecompression/m13real_rice.fits
fname = 'm13real_rice.fits'

f = pyfits.open(fname)
d = f[1].data
print 'before close f'
f.close()
print 'closed f'
```

Using astropy-1.1.1, this works, but with astropy-1.1.2, I get the following output:
```
before close f
Exception AttributeError: AttributeError('_coldefs',) in <bound method FITS_rec.__del__ of FITS_rec([(array([277,   0], dtype=int32), 0.0926625, 111.83617401123047),
       (array([314, 277], dtype=int32), 0.0699219339609146, 110.81782531738281),
       (array([316, 591], dtype=int32), 0.0926625, 110.81782531738281),
       (array([326, 907], dtype=int32), 0.0926625, 110.836181640625),
       (array([ 328, 1233], dtype=int32), 0.09351261792182922, 110.836181640625),
       (array([ 328, 1561], dtype=int32), 0.09351261792182922, 111.836181640625),
       (array([ 321, 1889], dtype=int32), 0.0926625, 111.836181640625),
       (array([ 307, 2210], dtype=int32), 0.09436273584365844, 111.83617401123047),
       (array([ 297, 2517], dtype=int32), 0.09436273584365844, 111.83617401123047),
       (array([ 295, 2814], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 289, 3109], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 290, 3398], dtype=int32), 0.0926625, 110.836181640625),
       (array([ 277, 3688], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 274, 3965], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 276, 4239], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 283, 4515], dtype=int32), 0.0926625, 111.836181640625),
       (array([ 290, 4798], dtype=int32), 0.0926625, 112.836181640625),
       (array([ 284, 5088], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 286, 5372], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 278, 5658], dtype=int32), 0.0926625, 112.836181640625),
       (array([ 277, 5936], dtype=int32), 0.0926625, 111.836181640625),
       (array([ 270, 6213], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 272, 6483], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 284, 6755], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 288, 7039], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 285, 7327], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 280, 7612], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 273, 7892], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 276, 8165], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 291, 8441], dtype=int32), 0.0926625, 112.836181640625),
       (array([ 301, 8732], dtype=int32), 0.0926625, 111.836181640625),
       (array([ 303, 9033], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 297, 9336], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 293, 9633], dtype=int32), 0.0926625, 111.83617401123047),
       (array([ 277, 9926], dtype=int32), 0.0926625, 111.83617401123047),
       (array([  265, 10203], dtype=int32), 0.0926625, 111.83617401123047),
       (array([  259, 10468], dtype=int32), 0.0926625, 111.83617401123047),
       (array([  264, 10727], dtype=int32), 0.0926625, 111.836181640625),
       (array([  268, 10991], dtype=int32), 0.0926625, 111.836181640625),
       (array([  271, 11259], dtype=int32), 0.0926625, 110.81782531738281),
       (array([  273, 11530], dtype=int32), 0.0926625, 110.81782531738281),
       (array([  267, 11803], dtype=int32), 0.0926625, 110.81782531738281),
       (array([  264, 12070], dtype=int32), 0.0926625, 110.81782531738281),
       (array([  264, 12334], dtype=int32), 0.0926625, 111.836181640625),
       (array([  259, 12598], dtype=int32), 0.09436273584365844, 111.836181640625),
       (array([  265, 12857], dtype=int32), 0.0926625, 112.85452270507812),
       (array([  271, 13122], dtype=int32), 0.0926625, 112.85452270507812),
       (array([  253, 13393], dtype=int32), 0.185325, 112.836181640625),
       (array([  272, 13646], dtype=int32), 0.185325, 112.836181640625),
       (array([  281, 13918], dtype=int32), 0.185325, 111.83617401123047),
       (array([  285, 14199], dtype=int32), 0.185325, 111.83617401123047),
       (array([  290, 14484], dtype=int32), 0.185325, 111.83617401123047),
       (array([  291, 14774], dtype=int32), 0.185325, 111.83617401123047),
       (array([  308, 15065], dtype=int32), 0.09351261792182922, 112.836181640625),
       (array([  284, 15373], dtype=int32), 0.0926625, 112.836181640625),
       (array([  269, 15657], dtype=int32), 0.0926625, 111.83617401123047),
       (array([  265, 15926], dtype=int32), 0.0926625, 111.83617401123047),
       (array([  288, 16191], dtype=int32), 0.09351261792182922, 110.81782531738281),
       (array([  285, 16479], dtype=int32), 0.1398438679218292, 110.81782531738281),
       (array([  286, 16764], dtype=int32), 0.185325, 112.85452270507812),
       (array([  290, 17050], dtype=int32), 0.185325, 112.85452270507812),
       (array([  289, 17340], dtype=int32), 0.185325, 112.836181640625),
       (array([  282, 17629], dtype=int32), 0.185325, 112.836181640625),
       (array([  276, 17911], dtype=int32), 0.185325, 111.836181640625),
       (array([  274, 18187], dtype=int32), 0.185325, 111.836181640625),
       (array([  278, 18461], dtype=int32), 0.185325, 111.83617401123047),
       (array([  279, 18739], dtype=int32), 0.185325, 111.83617401123047),
       (array([  318, 19018], dtype=int32), 0.0926625, 112.836181640625),
       (array([  282, 19336], dtype=int32), 0.185325, 112.836181640625),
       (array([  284, 19618], dtype=int32), 0.185325, 112.85452270507812),
       (array([  286, 19902], dtype=int32), 0.185325, 112.85452270507812),
       (array([  294, 20188], dtype=int32), 0.185325, 111.836181640625),
       (array([  295, 20482], dtype=int32), 0.185325, 111.836181640625),
       (array([  282, 20777], dtype=int32), 0.185325, 112.836181640625),
       (array([  276, 21059], dtype=int32), 0.185325, 112.836181640625),
       (array([  270, 21335], dtype=int32), 0.185325, 111.83617401123047),
       (array([  266, 21605], dtype=int32), 0.185325, 111.83617401123047),
       (array([  270, 21871], dtype=int32), 0.185325, 112.85452270507812),
       (array([  278, 22141], dtype=int32), 0.185325, 112.85452270507812),
       (array([  287, 22419], dtype=int32), 0.185325, 113.836181640625),
       (array([  296, 22706], dtype=int32), 0.185325, 113.836181640625),
       (array([  301, 23002], dtype=int32), 0.185325, 112.836181640625),
       (array([  300, 23303], dtype=int32), 0.185325, 112.836181640625),
       (array([  288, 23603], dtype=int32), 0.185325, 113.836181640625),
       (array([  275, 23891], dtype=int32), 0.185325, 113.836181640625),
       (array([  274, 24166], dtype=int32), 0.185325, 112.836181640625),
       (array([  294, 24440], dtype=int32), 0.185325, 113.836181640625),
       (array([  316, 24734], dtype=int32), 0.2123161556482315, 112.85452270507812),
       (array([  321, 25050], dtype=int32), 0.2779875, 112.85452270507812),
       (array([  310, 25371], dtype=int32), 0.37065, 113.836181640625),
       (array([  299, 25681], dtype=int32), 0.37065, 113.836181640625),
       (array([  302, 25980], dtype=int32), 0.23335648584365842, 112.836181640625),
       (array([  292, 26282], dtype=int32), 0.185325, 112.836181640625),
       (array([  279, 26574], dtype=int32), 0.185325, 112.836181640625),
       (array([  281, 26853], dtype=int32), 0.18702523584365843, 112.836181640625),
       (array([  289, 27134], dtype=int32), 0.185325, 113.836181640625),
       (array([  291, 27423], dtype=int32), 0.185325, 113.836181640625),
       (array([  285, 27714], dtype=int32), 0.185325, 114.836181640625),
       (array([  280, 27999], dtype=int32), 0.185325, 114.836181640625),
       (array([  271, 28279], dtype=int32), 0.185325, 113.87287139892578),
       (array([  254, 28550], dtype=int32), 0.2779875, 113.87287139892578),
       (array([  298, 28804], dtype=int32), 0.185325, 112.85452270507812),
       (array([  272, 29102], dtype=int32), 0.37065, 112.85452270507812),
       (array([  302, 29374], dtype=int32), 0.23335648584365842, 112.85452270507812),
       (array([  298, 29676], dtype=int32), 0.23335648584365842, 112.85452270507812),
       (array([  304, 29974], dtype=int32), 0.185325, 113.836181640625),
       (array([  285, 30278], dtype=int32), 0.23335648584365842, 113.836181640625),
       (array([  282, 30563], dtype=int32), 0.185325, 112.836181640625),
       (array([  281, 30845], dtype=int32), 0.185325, 112.836181640625),
       (array([  283, 31126], dtype=int32), 0.18702523584365843, 113.836181640625),
       (array([  285, 31409], dtype=int32), 0.185325, 113.836181640625),
       (array([  290, 31694], dtype=int32), 0.18872547168731688, 112.85452270507812),
       (array([  276, 31984], dtype=int32), 0.2805378537654877, 112.85452270507812),
       (array([  270, 32260], dtype=int32), 0.37065, 113.836181640625),
       (array([  279, 32530], dtype=int32), 0.37065, 113.836181640625),
       (array([  275, 32809], dtype=int32), 0.42038172168731686, 112.85452270507812),
       (array([  251, 33084], dtype=int32), 0.7413, 112.85452270507812),
       (array([  268, 33335], dtype=int32), 0.555975, 114.836181640625),
       (array([  275, 33603], dtype=int32), 0.37405047168731687, 114.836181640625),
       (array([  265, 33878], dtype=int32), 0.37065, 114.836181640625),
       (array([  249, 34143], dtype=int32), 0.50964375, 114.836181640625),
       (array([  265, 34392], dtype=int32), 0.50964375, 112.85452270507812),
       (array([  281, 34657], dtype=int32), 0.46331249999999996, 114.836181640625),
       (array([  293, 34938], dtype=int32), 0.37065, 114.836181640625),
       (array([  285, 35231], dtype=int32), 0.37065, 114.836181640625),
       (array([  273, 35516], dtype=int32), 0.37065, 113.836181640625),
       (array([  272, 35789], dtype=int32), 0.37065, 113.836181640625),
       (array([  291, 36061], dtype=int32), 0.37065, 113.836181640625),
       (array([  305, 36352], dtype=int32), 0.37065, 113.836181640625),
       (array([  307, 36657], dtype=int32), 0.37745094337463375, 113.87287139892578),
       (array([  302, 36964], dtype=int32), 0.3740497647285461, 113.87287139892578),
       (array([  285, 37266], dtype=int32), 0.42038172168731686, 113.87287139892578),
       (array([  295, 37551], dtype=int32), 0.32686910376548767, 113.87287139892578),
       (array([  308, 37846], dtype=int32), 0.2796877358436584, 114.836181640625),
       (array([  281, 38154], dtype=int32), 0.555975, 114.89122009277344),
       (array([  287, 38435], dtype=int32), 0.555975, 114.836181640625),
       (array([  274, 38722], dtype=int32), 0.6949687499999999, 114.836181640625),
       (array([  283, 38996], dtype=int32), 0.555975, 113.87287139892578),
       (array([  293, 39279], dtype=int32), 0.41698124999999997, 113.87287139892578),
       (array([  296, 39572], dtype=int32), 0.32686910376548767, 114.89122009277344),
       (array([  303, 39868], dtype=int32), 0.2779875, 114.89122009277344),
       (array([  316, 40171], dtype=int32), 0.23335648584365842, 114.836181640625),
       (array([  296, 40487], dtype=int32), 0.37065, 114.836181640625),
       (array([  300, 40783], dtype=int32), 0.37065, 114.836181640625),
       (array([  293, 41083], dtype=int32), 0.42038172168731686, 114.836181640625),
       (array([  295, 41376], dtype=int32), 0.37065, 113.836181640625),
       (array([  288, 41671], dtype=int32), 0.37065, 113.836181640625),
       (array([  300, 41959], dtype=int32), 0.25694716980457305, 113.836181640625),
       (array([  282, 42259], dtype=int32), 0.37065, 113.836181640625),
       (array([  266, 42541], dtype=int32), 0.50964375, 113.836181640625),
       (array([  271, 42807], dtype=int32), 0.41698124999999997, 113.836181640625),
       (array([  280, 43078], dtype=int32), 0.28053820724487305, 112.836181640625),
       (array([  300, 43358], dtype=int32), 0.18702523584365843, 112.836181640625),
       (array([  297, 43658], dtype=int32), 0.18872547168731688, 113.87287139892578),
       (array([  300, 43955], dtype=int32), 0.18872547168731688, 113.87287139892578),
       (array([  279, 44255], dtype=int32), 0.2796877358436584, 114.836181640625),
       (array([  277, 44534], dtype=int32), 0.2805375002861023, 114.836181640625),
       (array([  280, 44811], dtype=int32), 0.2805375002861023, 112.85452270507812),
       (array([  278, 45091], dtype=int32), 0.32431875, 113.836181640625),
       (array([  305, 45369], dtype=int32), 0.23335648584365842, 113.836181640625),
       (array([  302, 45674], dtype=int32), 0.2779875, 113.836181640625),
       (array([  295, 45976], dtype=int32), 0.37065, 113.87287139892578),
       (array([  302, 46271], dtype=int32), 0.37405047168731687, 113.87287139892578),
       (array([  295, 46573], dtype=int32), 0.41698124999999997, 113.836181640625),
       (array([  263, 46868], dtype=int32), 0.6486375, 113.836181640625),
       (array([  262, 47131], dtype=int32), 0.5138943396091461, 113.87287139892578),
       (array([  273, 47393], dtype=int32), 0.37065, 113.87287139892578),
       (array([  280, 47666], dtype=int32), 0.37065, 113.836181640625),
       (array([  286, 47946], dtype=int32), 0.37065, 113.836181640625),
       (array([  281, 48232], dtype=int32), 0.42038172168731686, 112.836181640625),
       (array([  288, 48513], dtype=int32), 0.37065, 112.836181640625),
       (array([  300, 48801], dtype=int32), 0.2805378537654877, 113.836181640625),
       (array([  294, 49101], dtype=int32), 0.32686910376548767, 113.836181640625),
       (array([  321, 49395], dtype=int32), 0.18872547168731688, 114.836181640625),
       (array([  288, 49716], dtype=int32), 0.32686910376548767, 114.836181640625),
       (array([  278, 50004], dtype=int32), 0.37065, 113.836181640625),
       (array([  263, 50282], dtype=int32), 0.46331249999999996, 113.836181640625),
       (array([  275, 50545], dtype=int32), 0.37065, 114.836181640625),
       (array([  255, 50820], dtype=int32), 0.46331249999999996, 114.89122009277344),
       (array([  248, 51075], dtype=int32), 0.46331249999999996, 114.89122009277344),
       (array([  262, 51323], dtype=int32), 0.37065, 113.87287139892578),
       (array([  273, 51585], dtype=int32), 0.2805378537654877, 113.87287139892578),
       (array([  291, 51858], dtype=int32), 0.18872547168731688, 113.87287139892578),
       (array([  291, 52149], dtype=int32), 0.18872547168731688, 114.89122009277344),
       (array([  272, 52440], dtype=int32), 0.2830878540515899, 113.87287139892578),
       (array([  263, 52712], dtype=int32), 0.32431875, 112.85452270507812),
       (array([  291, 52975], dtype=int32), 0.18702523584365843, 112.85452270507812),
       (array([  286, 53266], dtype=int32), 0.185325, 113.87287139892578),
       (array([  282, 53552], dtype=int32), 0.185325, 113.87287139892578),
       (array([  289, 53834], dtype=int32), 0.18872547168731688, 113.836181640625),
       (array([  283, 54123], dtype=int32), 0.2779875, 113.836181640625),
       (array([  283, 54406], dtype=int32), 0.37065, 112.836181640625),
       (array([  284, 54689], dtype=int32), 0.3740497647285461, 112.836181640625),
       (array([  277, 54973], dtype=int32), 0.37065, 111.836181640625),
       (array([  263, 55250], dtype=int32), 0.37065, 111.836181640625),
       (array([  288, 55513], dtype=int32), 0.185325, 113.836181640625),
       (array([  277, 55801], dtype=int32), 0.185325, 113.836181640625),
       (array([  275, 56078], dtype=int32), 0.185325, 113.836181640625),
       (array([  288, 56353], dtype=int32), 0.185325, 113.836181640625),
       (array([  305, 56641], dtype=int32), 0.185325, 113.87287139892578),
       (array([  295, 56946], dtype=int32), 0.2779875, 113.87287139892578),
       (array([  326, 57241], dtype=int32), 0.18872547168731688, 113.836181640625),
       (array([  329, 57567], dtype=int32), 0.18872547168731688, 113.836181640625),
       (array([  288, 57896], dtype=int32), 0.37065, 112.85452270507812),
       (array([  279, 58184], dtype=int32), 0.32431875, 112.85452270507812),
       (array([  286, 58463], dtype=int32), 0.18702523584365843, 112.836181640625),
       (array([  267, 58749], dtype=int32), 0.185325, 112.836181640625),
       (array([  260, 59016], dtype=int32), 0.185325, 112.85452270507812),
       (array([  259, 59276], dtype=int32), 0.185325, 111.836181640625),
       (array([  272, 59535], dtype=int32), 0.185325, 109.836181640625),
       (array([  280, 59807], dtype=int32), 0.185325, 109.836181640625),
       (array([  279, 60087], dtype=int32), 0.185325, 110.836181640625),
       (array([  277, 60366], dtype=int32), 0.18702523584365843, 110.836181640625),
       (array([  275, 60643], dtype=int32), 0.185325, 112.836181640625),
       (array([  274, 60918], dtype=int32), 0.185325, 112.836181640625),
       (array([  280, 61192], dtype=int32), 0.185325, 111.836181640625),
       (array([  281, 61472], dtype=int32), 0.185325, 111.836181640625),
       (array([  285, 61753], dtype=int32), 0.185325, 112.836181640625),
       (array([  283, 62038], dtype=int32), 0.185325, 112.836181640625),
       (array([  283, 62321], dtype=int32), 0.185325, 112.836181640625),
       (array([  285, 62604], dtype=int32), 0.185325, 112.836181640625),
       (array([  294, 62889], dtype=int32), 0.185325, 112.85452270507812),
       (array([  301, 63183], dtype=int32), 0.185325, 112.85452270507812),
       (array([  282, 63484], dtype=int32), 0.2779875, 112.836181640625),
       (array([  299, 63766], dtype=int32), 0.185325, 113.836181640625),
       (array([  295, 64065], dtype=int32), 0.185325, 111.83617401123047),
       (array([  305, 64360], dtype=int32), 0.185325, 111.83617401123047),
       (array([  317, 64665], dtype=int32), 0.185325, 112.85452270507812),
       (array([  318, 64982], dtype=int32), 0.185325, 112.85452270507812),
       (array([  311, 65300], dtype=int32), 0.185325, 111.836181640625),
       (array([  298, 65611], dtype=int32), 0.185325, 111.836181640625),
       (array([  282, 65909], dtype=int32), 0.185325, 111.836181640625),
       (array([  267, 66191], dtype=int32), 0.185325, 111.836181640625),
       (array([  256, 66458], dtype=int32), 0.185325, 112.836181640625),
       (array([  248, 66714], dtype=int32), 0.185325, 112.836181640625),
       (array([  282, 66962], dtype=int32), 0.09436273584365844, 112.85452270507812),
       (array([  287, 67244], dtype=int32), 0.09436273584365844, 112.85452270507812),
       (array([  297, 67531], dtype=int32), 0.09436273584365844, 110.836181640625),
       (array([  271, 67828], dtype=int32), 0.185325, 110.836181640625),
       (array([  272, 68099], dtype=int32), 0.185325, 110.836181640625),
       (array([  270, 68371], dtype=int32), 0.185325, 110.836181640625),
       (array([  273, 68641], dtype=int32), 0.185325, 112.836181640625),
       (array([  275, 68914], dtype=int32), 0.185325, 112.836181640625),
       (array([  286, 69189], dtype=int32), 0.13984422140121458, 110.836181640625),
       (array([  267, 69475], dtype=int32), 0.185325, 110.836181640625),
       (array([  259, 69742], dtype=int32), 0.185325, 111.836181640625),
       (array([  267, 70001], dtype=int32), 0.1398438679218292, 111.836181640625),
       (array([  286, 70268], dtype=int32), 0.0926625, 111.83617401123047),
       (array([  283, 70554], dtype=int32), 0.0926625, 111.83617401123047),
       (array([  276, 70837], dtype=int32), 0.0926625, 110.836181640625),
       (array([  273, 71113], dtype=int32), 0.0926625, 110.836181640625),
       (array([  270, 71386], dtype=int32), 0.09351261792182922, 110.836181640625),
       (array([  267, 71656], dtype=int32), 0.0926625, 110.836181640625),
       (array([  266, 71923], dtype=int32), 0.0926625, 111.83617401123047),
       (array([  263, 72189], dtype=int32), 0.0926625, 111.83617401123047),
       (array([  259, 72452], dtype=int32), 0.0926625, 111.836181640625),
       (array([  251, 72711], dtype=int32), 0.0926625, 111.836181640625),
       (array([  254, 72962], dtype=int32), 0.0926625, 111.83617401123047),
       (array([  253, 73216], dtype=int32), 0.0926625, 111.83617401123047),
       (array([  249, 73469], dtype=int32), 0.0926625, 111.83617401123047),
       (array([  250, 73718], dtype=int32), 0.0926625, 111.83617401123047),
       (array([  254, 73968], dtype=int32), 0.0926625, 111.836181640625),
       (array([  261, 74222], dtype=int32), 0.0926625, 111.836181640625),
       (array([  214, 74483], dtype=int32), 0.25100032065542327, 111.836181640625),
       (array([  212, 74697], dtype=int32), 0.28863560732668503, 111.836181640625),
       (array([  276, 74909], dtype=int32), 0.0926625, 111.83617401123047),
       (array([  286, 75185], dtype=int32), 0.0926625, 112.836181640625),
       (array([  298, 75471], dtype=int32), 0.0926625, 111.836181640625),
       (array([  308, 75769], dtype=int32), 0.09436273584365844, 111.836181640625),
       (array([  309, 76077], dtype=int32), 0.0926625, 111.836181640625),
       (array([  306, 76386], dtype=int32), 0.0926625, 111.836181640625),
       (array([  291, 76692], dtype=int32), 0.0926625, 110.836181640625),
       (array([  273, 76983], dtype=int32), 0.0926625, 110.836181640625),
       (array([  273, 77256], dtype=int32), 0.0926625, 109.836181640625),
       (array([  286, 77529], dtype=int32), 0.0926625, 109.836181640625),
       (array([  299, 77815], dtype=int32), 0.0926625, 110.836181640625),
       (array([  302, 78114], dtype=int32), 0.0926625, 110.836181640625),
       (array([  296, 78416], dtype=int32), 0.0926625, 111.836181640625),
       (array([  284, 78712], dtype=int32), 0.0926625, 111.836181640625),
       (array([  270, 78996], dtype=int32), 0.0926625, 111.836181640625),
       (array([  257, 79266], dtype=int32), 0.0926625, 111.836181640625),
       (array([  252, 79523], dtype=int32), 0.0926625, 110.836181640625),
       (array([  259, 79775], dtype=int32), 0.0926625, 110.836181640625),
       (array([  272, 80034], dtype=int32), 0.0926625, 110.836181640625),
       (array([  282, 80306], dtype=int32), 0.0926625, 110.836181640625),
       (array([  224, 80588], dtype=int32), 0.2578451361001941, 110.836181640625),
       (array([  231, 80812], dtype=int32), 0.2023181733913942, 111.83617401123047),
       (array([  275, 81043], dtype=int32), 0.0926625, 110.836181640625),
       (array([  280, 81318], dtype=int32), 0.0926625, 110.836181640625),
       (array([  283, 81598], dtype=int32), 0.0926625, 110.836181640625),
       (array([  280, 81881], dtype=int32), 0.0926625, 110.836181640625),
       (array([  278, 82161], dtype=int32), 0.0926625, 109.836181640625),
       (array([  271, 82439], dtype=int32), 0.0926625, 109.836181640625),
       (array([  264, 82710], dtype=int32), 0.0926625, 109.836181640625),
       (array([  254, 82974], dtype=int32), 0.0926625, 109.836181640625),
       (array([  228, 83228], dtype=int32), 0.14637073640437456, 110.836181640625),
       (array([  205, 83456], dtype=int32), 0.27372067173094833, 110.836181640625),
       (array([  205, 83661], dtype=int32), 0.35533014593762313, 109.836181640625),
       (array([  196, 83866], dtype=int32), 0.4991649365492337, 109.836181640625),
       (array([  290, 84062], dtype=int32), 0.0926625, 109.836181640625)], 
      dtype=(numpy.record, [('COMPRESSED_DATA', '>i4', (2,)), ('ZSCALE', '>f8'), ('ZZERO', '>f8')]))> ignored
closed f
```

Please fix this.  Thank you!

### PR DESCRIPTION
**Pull request by pllim**
Fixes #4690 

Closes #4694 

@astrofrog , this is a rebased version of #4694 for the release. I found that I needed this for my project, so might as well...

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -655,6 +655,9 @@ Bug Fixes
   - Fix convenience functions (``getdata``, ``getheader``, ``append``,
     ``update``) to close files. [#4786]
 
+  - Removed raising of AssertionError that could occur after closing or
```
Comment (pllim) [inline CHANGES.rst]: This is under 1.1.3 as per original PR. But I can't find that milestone tag anymore. Should I move this to 1.2?

Comment (astrofrog) [inline CHANGES.rst]: Yes, 1.1.3 won't happen.

Comment (pllim) [inline CHANGES.rst]: Okay. Moved and squashed.

**Comment by mhvk** [comment]
Everyone was already happy with this before, so merging. Thanks, @pllim, @rirze

**Comment by astrofrog** [comment]
I'm going to backport this to v1.0.x

**Comment by pllim** [comment]
Code-wise, this is superseded by #4977. But it doesn't hurt to backport this one as well.

**Comment by astrofrog** [comment]
@pllim - indeed, but #4977 applied cleanly only if this fix was also backported, so it was easier to do both.

**Pull request merged**